src/main/kotlin/no/nav/amt/deltaker/bff/deltaker/DeltakerService.kt

```diff
@@ -33,96 +29,48 @@ class DeltakerService(
-                sluttdato = endring.sluttdato,
-            )
-
-            is DeltakerEndring.Endring.IkkeAktuell -> opprinneligDeltaker.copy(
-                status = nyDeltakerStatus(DeltakerStatus.Type.IKKE_AKTUELL, endring.aarsak?.toDeltakerStatusAarsak()),
-            )
-
-            is DeltakerEndring.Endring.ForlengDeltakelse -> {
-                if (opprinneligDeltaker.status.type == DeltakerStatus.Type.HAR_SLUTTET &&
-                    endring.sluttdato.isAfter(LocalDate.now())
-                ) {
-                    opprinneligDeltaker.copy(
-                        status = nyDeltakerStatus(DeltakerStatus.Type.DELTAR),
-                        sluttdato = endring.sluttdato,
-                    )
-                } else {
-                    opprinneligDeltaker.copy(
-                        sluttdato = endring.sluttdato,
-                    )
-                }
-            }
-
-            is DeltakerEndring.Endring.AvsluttDeltakelse -> opprinneligDeltaker.copy(
-                status = nyDeltakerStatus(DeltakerStatus.Type.HAR_SLUTTET, endring.aarsak.toDeltakerStatusAarsak()),
-                sluttdato = endring.sluttdato,
-            )
-
-            is DeltakerEndring.Endring.EndreSluttarsak -> opprinneligDeltaker.copy(
-                status = nyDeltakerStatus(DeltakerStatus.Type.HAR_SLUTTET, endring.aarsak.toDeltakerStatusAarsak()),
-            )
+            is DeltakerEndring.Endring.EndreBakgrunnsinformasjon ->
+                endreBakgrunnsinformasjon(opprinneligDeltaker, endretAv, endretAvEnhet, endring)
+
+            is DeltakerEndring.Endring.AvsluttDeltakelse -> TODO()
+            is DeltakerEndring.Endring.EndreDeltakelsesmengde -> TODO()
+            is DeltakerEndring.Endring.EndreInnhold -> TODO()
+            is DeltakerEndring.Endring.EndreSluttarsak -> TODO()
+            is DeltakerEndring.Endring.EndreSluttdato -> TODO()
+            is DeltakerEndring.Endring.EndreStartdato -> TODO()
+            is DeltakerEndring.Endring.ForlengDeltakelse -> TODO()
+            is DeltakerEndring.Endring.IkkeAktuell -> TODO()
         }
 
-        if (erEndret(opprinneligDeltaker, deltaker)) {
-            val navAnsatt = navAnsattService.hentEllerOpprettNavAnsatt(endretAv)
-            val navEnhet = endretAvEnhet.let { navEnhetService.hentEllerOpprettNavEnhet(it) }
-            val deltakerEndring = DeltakerEndring(
-                id = UUID.randomUUID(),
-                deltakerId = deltaker.id,
-                endring = endring,
-                endretAv = navAnsatt.id,
-                endretAvEnhet = navEnhet.id,
-                endret = LocalDateTime.now(),
-            )
-            upsert(deltaker.copy(historikk = deltaker.historikk.plus(DeltakerHistorikk.Endring(deltakerEndring))))
-            log.info("Oppdatert deltaker med id ${deltaker.id}")
-        }
-        return deltakerRepository.get(deltaker.id).getOrThrow()
+        return deltaker
```

Comment:
Litt usikker på hva som er best å gjøre her. I dag så bruker frontend resultatet av endringen til å oppdatere frontend, så for nå så returnerer jeg bare en "dum" deltaker tilbake her. Endringen har jo ikke feilet i amt-deltaker så det er jo kanskje godt nok.

Jeg synes kanskje frontend burde bare få et 200 fra apiet her, også gjøre en get selv, men det vil jo alltid være en liten risiko for at vi ikke har rukket å lese siste kafkamelding.

Reply:
Jeg er fortsatt veldig usikker på om det er godt nok å kun oppdatere via kafka.. Jeg synes iallefall det du gjør her er en fin løsning frem til vi lærer noe annet :)